### ISSUE TITLE
[BUG/UI] Using the code editor is practically impossible on mobile

### ISSUE THREAD

**Issue by Reinitialized** [go-gitea/gitea#14776]
As a user and programmer using Gitea, it is impossible in practice to utilize the built in code editor for making quick changes on a file. Attempting to scroll through the editor results in a "right click" menu to pop up displaying keyboard shortcuts as if I'm using a mouse/keyboard, and deleting text does not seem to register at all.

IMO, Gitea is intended to be an extended toolset on top of git, making the UX of the git toolset easier to use. I think the current editor implementation is out of scope and feature bloat, and propose replacing such along the lines of:

- textarea
- shows line numbers on the left

Gitea should not be a replacement for an IDE. For reference sake, as this classifies as a separate issue nor have I tested such tools myself, Gitea should ensure it is able to cleanly integrate with existing tools such as GitPod or Theia, not implement a "half baked" alternative as currently existing. Editing files on Gitea should be for "quick fixes", such as typos or something along the line of only taking a minute or two to do.

**Comment by phtan** [comment]
I have had a similar problem with the menu that appears on "right click" (or "long press" on mobile (Android)).

When I want to paste some code/text into the text-editor (of Gitea) - while using a mobile phone - I do a long press. I expected an option of "paste" to appear. But I saw a menu that features keyboard shortcuts.

This situation obstructs me from pasting. This is ususual behaviour for software that runs on mobile phones, to put things mildly.

**Comment by phtan** [comment]
I wonder if changing the below-hyperlinked line of code gets rid of the offending "right-click menu"?

Specifically, changing "activate" to "nothing" 🤔

I have no development environment right now to test if the above change indeed succeeds. (I.e. no laptop nor desktop Personal Computer for now, just a mobile phone). I'm just going to make a suggestion here, in case someone else can implement this suggestion (and possibly make a Pull Request).

https://github.com/go-gitea/gitea/blob/5d2b7ba63958a4eee57c0a5300f20e9df6f32c37/web_src/js/vendor/semanticdropdown.js#L3973

**Comment by Username404-59** [comment]
I can confirm this, the editor is a bit broken on mobile; typing characters seems to work (with latency?) but removing them does not.

**Comment by Thewisem** [comment]
I second this. Its a pain to do anything in that text editor

**Comment by fairking** [comment]
How am I supposed to create a new file??? 
There is no button. :-(

![Screenshot_20220527-220206](https://user-images.githubusercontent.com/13495631/170789303-80b88f8a-4139-48ec-8e4f-8273b045c2fa.png)

**Comment by HackerDaGreat57** [comment]
@fairking first you have to add at least one file via command line Git only then you can start adding files via web.

**Comment by redthing1** [comment]
This editor is endlessly frustrating on mobile. I hope we can resolve it. Maybe even serving a different editor for a mobile device?

**Comment by silverwind** [comment]
There were some recent improvements in Monaco related to Mobile. Try in latest gitea version.

**Comment by Forza-tng** [comment]
Hi. I came here too because editing files via mobile UI is mostly impossible. I don't think a lot needs to be adjusted to make it work and it would be of great benefit.

- cant select text
- cant paste
- difficult to scroll - it moves back to top
- cant see all text because keyboard covers.

I'm using Gitea 1.21.2 on Gentoo Linux

My workaround is to do ssh from my phone, make changes in nano and the git push... Needless to say this isn't very efficient.

[Screen_Recording_20231217_195224_Opera.webm](https://github.com/go-gitea/gitea/assets/68693597/0bedcdfc-4eb0-457c-beb0-61ff69fc842a)

**Comment by duchenpaul** [comment]
Yes, can we just have a toggle in the config file that disables this editor to make the code editor as a giant text box?

This code editor is stupid.

**Comment by delvh** [comment]
#30474 :grin:

**Comment by wxiaoguang** [comment]
> #30474 😁

`30474` is for "markdown editor" (for issue/comment/wiki content editing)

This issue's problem is the "code editor": currently Monaco Editor on the "Edit File" page.
